frontend/catalyst/device/qjit_device.py

```diff
@@ -84,6 +84,87 @@
+    "braket.aws.qubit": ("OpenQasmDevice", "librtd_openqasm"),
+    "braket.local.qubit": ("OpenQasmDevice", "librtd_openqasm"),
+}
+
+
+@dataclass
+class BackendInfo:
+    """Backend information"""
+
+    device_name: str
+    c_interface_name: str
+    lpath: str
+    kwargs: Dict[str, Any]
+
+
+def extract_backend_info(device: qml.QubitDevice, capabilities: DeviceCapabilities) -> BackendInfo:
+    """Extract the backend info from a quantum device. The device is expected to carry a reference
+    to a valid TOML config file."""
+    # pylint: disable=too-many-branches
+
+    dname = device.name
+    if isinstance(device, qml.Device):
+        dname = device.short_name
+
+    device_name = ""
+    device_lpath = ""
+    device_kwargs = {}
+
+    if dname in SUPPORTED_RT_DEVICES:
+        # Support backend devices without `get_c_interface`
+        device_name = SUPPORTED_RT_DEVICES[dname][0]
+        device_lpath = get_lib_path("runtime", "RUNTIME_LIB_DIR")
+        sys_platform = platform.system()
+
+        if sys_platform == "Linux":
+            device_lpath = os.path.join(device_lpath, SUPPORTED_RT_DEVICES[dname][1] + ".so")
+        elif sys_platform == "Darwin":  # pragma: no cover
+            device_lpath = os.path.join(device_lpath, SUPPORTED_RT_DEVICES[dname][1] + ".dylib")
+        else:  # pragma: no cover
+            raise NotImplementedError(f"Platform not supported: {sys_platform}")
+    elif hasattr(device, "get_c_interface"):
+        # Support third party devices with `get_c_interface`
+        device_name, device_lpath = device.get_c_interface()
+    else:
+        raise CompileError(f"The {dname} device does not provide C interface for compilation.")
+
+    if not pathlib.Path(device_lpath).is_file():
+        raise CompileError(f"Device at {device_lpath} cannot be found!")
+
+    if hasattr(device, "shots"):
+        if isinstance(device, qml.Device):
+            device_kwargs["shots"] = device.shots if device.shots else 0
+        else:
+            # TODO: support shot vectors
+            device_kwargs["shots"] = device.shots.total_shots if device.shots else 0
+
+    if dname == "braket.local.qubit":  # pragma: no cover
+        device_kwargs["device_type"] = dname
+        device_kwargs["backend"] = (
+            # pylint: disable=protected-access
+            device._device._delegate.DEVICE_ID
+        )
+    elif dname == "braket.aws.qubit":  # pragma: no cover
+        device_kwargs["device_type"] = dname
+        device_kwargs["device_arn"] = device._device._arn  # pylint: disable=protected-access
+        if device._s3_folder:  # pylint: disable=protected-access
+            device_kwargs["s3_destination_folder"] = str(
+                device._s3_folder  # pylint: disable=protected-access
+            )
+
+    for k, v in capabilities.options.items():
+        if hasattr(device, v):
+            device_kwargs[k] = getattr(device, v)
+
+    return BackendInfo(dname, device_name, device_lpath, device_kwargs)
+
```

Comment:
Popping in because I was initially going to ask this on #626, but realized that it's actually from here after the PR was rebased to this one. 

Is this an exact copy of the code in `utils.runtime.py`, or are there modifications?

Reply:
I don't think I changed the logic of BackendInfo loading, it should be the exact copy.

Reply:
Cool - should we either remove it from `utils.runtime.py`? Or import it from there? Is this to get around an import error?

Reply:
Oh, I see. I think something have messed up the pr. It was supposed to remove runtime.py. Will fix

Reply:
Runtime.py is removed now.